src/main/java/org/commonjava/util/partyline/lock/local/LocalLockOwner.java

```diff
@@ -182,36 +184,39 @@ synchronized boolean unlock( final String label )
                     this.dominantLockLevel = newDom.level;
                     logger.trace( "New dominant holder is: {} with level: {}", this.dominantOwner,
                                   this.dominantLockLevel );
+                    return new UnlockStatus( true, true, this.dominantLockLevel );
                 }
                 else
                 {
                     logger.trace( "Locks seems to be empty; Unlocking" );
                     this.dominantOwner = null;
                     this.dominantLockLevel = null;
+                    // TODO: how best to handle this case?
+                    return new UnlockStatus( true, true, null );
```

Comment:
I think this is correct...we had a lock, we don't now, therefore dominant lock has changed.